File: cuttingPolygonsPython/src/smartcutter.py
import src.polygon as polygon
import src.cutter as cutter


class SmartCutter:
    def __init__(self, points, polygonVertices, segment):
        type = polygon.Polygon(polygonVertices)
        if (type.typePolygon()):
            self._concave = cutter.ConcaveCutter()
            self._concave.cutPolygon(points, polygonVertices, segment)
        else:
            self._convex = cutter.ConvexCutter()
            self._convex.cutPolygon(points, polygonVertices, segment)

File: CuttingPolygonsC/src/smartcutter.cpp
#include "smartcutter.h"
namespace smartCutterNamespace {


SmartCutter::SmartCutter()
{

}

SmartCutter::SmartCutter(const vector<Point> &points, const vector<Vertex> &polygonVertices, const Segment &segment)
{
    Polygon tipo;
    tipo=Polygon(polygonVertices);
    if (tipo.TypePolygon())
    {
        _concavo=ConcaveCutter();
        _concavo.CutPolygon(points,polygonVertices,segment);
    }
    else
    {
        _convesso=ConvexCutter();
        _convesso.CutPolygon(points,polygonVertices,segment);
    }
}
}
